src/compiler/diagnosticMessages.json

```diff
@@ -1844,14 +1844,6 @@
         "category": "Error",
         "code": 2359
     },
-    "The left-hand side of an 'in' expression must be a private identifier or of type 'any', 'string', 'number', or 'symbol'.": {
-        "category": "Error",
-        "code": 2360
-    },
-    "The right-hand side of an 'in' expression must not be a primitive.": {

```

Comment:
I think losing these error messages is actually worse.

Reply:
I disagree. We'd tell you it can't be a primitive, but we wouldn't tell you _why_ it could be. There was never any elaboration. Now there is. The fact that the error applies to the right hand of an `in` expression is obvious from context so it just seems like useless information.